src/Silverback.Integration.Kafka/Messaging/Configuration/Kafka/ConfluentClientConfigProxies.cs

```diff
@@ -876,6 +876,17 @@ public int CancellationDelayMaxMs
     [SuppressMessage("", "CA1044", Justification = "Accessors generated according to wrapped class")]
     public abstract class ConfluentConsumerConfigProxy : ConfluentClientConfigProxy
     {
+        /// <summary>
+        ///     If the KafkaConsumer does not have a group.id for consumer group partition assignment and offset
+        ///     storage, then a fake value is set as workaround for the current limitations of confluent-kafka-dotnet and librdkafka.
+        ///     See also
+        ///     <see href="https://github.com/confluentinc/confluent-kafka-dotnet/issues/225" />
+        ///     <see href="https://github.com/edenhill/librdkafka/issues/593" /> for more information.
+        /// </summary>
+        public const string GroupIdNotSet = "not-set";
+
+        private bool? _isGroupIdSet;
```

Comment:
As mentioned (and as written in the comment at the top) this file is autogenerated via the [Silverback.Tools.KafkaConfigClassGenerator](https://github.com/BEagle1984/silverback/tree/master/tools/Silverback.Tools.KafkaConfigClassGenerator). The tool is used whenever a new version of `Confluent.Kafka` is linked, to adapt this proxy classes accordingly.
You can't therefore simply edit this file but you need to modify the generator.
My suggestion would be to generate the ClientId as abstract and implement it in [KafkaConsumerConfig](https://github.com/BEagle1984/silverback/blob/master/src/Silverback.Integration.Kafka/Messaging/Configuration/Kafka/KafkaConsumerConfig.cs) (moving the constant in there as well).

Reply:
Does it need to be public for some reason?

Reply:
Thanks for the hint, I moved the property to KafkaConsumerConfig:
https://github.com/BEagle1984/silverback/blob/491926231cbcd0b64afe9452e05c97e0fe55ff05/src/Silverback.Integration.Kafka/Messaging/Configuration/Kafka/KafkaConsumerConfig.cs#L53

And GroupId is now abstract:
https://github.com/BEagle1984/silverback/blob/491926231cbcd0b64afe9452e05c97e0fe55ff05/src/Silverback.Integration.Kafka/Messaging/Configuration/Kafka/ConfluentClientConfigProxies.cs#L926